authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.authorization.utils;
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AlternativeAuthorizationResourceActionProvider;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.service.CommonPermissionCheckingUtils;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+
+@Component
+public class EventAuthorizationUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventAuthorizationUtils.class);
+
+    private AlternativeAuthorizationResourceActionProvider actionAlternativeProvider;
+
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
+    public EventAuthorizationUtils(CommonPermissionCheckingUtils permissionCheckingUtils,
+            AlternativeAuthorizationResourceActionProvider actionAlternativeProvider) {
+        this.actionAlternativeProvider = actionAlternativeProvider;
+        this.permissionCheckingUtils = permissionCheckingUtils;
+    }
+
+    public void checkPermissionBasedOnResourceTypeAndCrn(Collection<EventAuthorizationDto> eventAuthorizationDtos) {
+        throwIfNull(eventAuthorizationDtos,
+                () -> new IllegalArgumentException("The collection of " + EventAuthorizationDto.class.getSimpleName() + "s should not be null!"));
+        for (EventAuthorizationDto dto : eventAuthorizationDtos) {
+            String resourceType = dto.getResourceType();
+            Arrays.asList(AuthorizationResourceAction.values()).stream()
+                    .filter(action -> isResourceTypeHasDescribeOrGetAction(action, resourceType))
+                    .findFirst()
+                    .ifPresentOrElse(
```

Comment:
So if the user doesn't have permission for one resource in one hundred than we throw an exception. Is this the expected behaviour?

Reply:
We didn't have any expectations/DoD in the first place 😄 I mean nobody has before, but it was my first thought, because if we give an incomplete set of events (only those that the user has rights over) that could cause hard times when some issue occurs and one would like to investigate it based on the events.
But I'm open for discussion over it, this is the time for reasons and counter-reasons 🙂